### PR TITLE
Rename deprecated environment variables in test Makefiles

### LINKED ISSUE #4127: failure due to deprecated MODULE in testcase Makefile
Running `make test` fails due to error in `cocotb/tests/test_cases/issue_253/Makefile` as it uses deprecated MODULE not COCOTB_TEST_MODULE

>         raise RuntimeError(
>                                                         RuntimeError: Environment variable COCOTB_TEST_MODULES, which defines the module(s) to execute, is not defined or empty.
> 

Replacing MODULE with COCOTB_TEST_MODULE works fine

### PR DESCRIPTION
**Pull request by dolphingarlic**
* Rename `MODULE` to `COCOTB_TEST_MODULES`
* Rename `TOPLEVEL` to `COCOTB_TOPLEVEL`

closes #4127

### CODE REVIEW

**Review by ktbarrett — APPROVED** [review]

**Review by imphil — COMMENTED** [review]
Thanks for your contribution! 

Your PR description says you're renaming `MODULE` only, but the change also renames some (but not all) `TOPLEVEL` variables. Would you mind either renaming all the rest of the `TOPLEVEL` variables as well and adjust the commit message, or alternatively remove the `TOPLEVEL` renames from the commit?

File: README.md [hunk 1]
```diff
@@ -106,8 +106,8 @@ A simple Makefile:
 
 TOPLEVEL_LANG = verilog
 VERILOG_SOURCES = $(shell pwd)/dff.sv
-TOPLEVEL = dff
-MODULE = test_dff
+COCOTB_TOPLEVEL = dff
+COCOTB_TEST_MODULES = test_dff
```
Comment (imphil) [inline README.md:110]: Thanks! We are keeping the README.md compatible with the current stable release (since people got confused otherwise). So can you please revert this change in the README?

**Review by imphil — COMMENTED** [review]
Thanks a lot for your updates!

One small change needed to the README, and then we're ready to merge.

File: README.md (hunk 1 above)
Comment (dolphingarlic) [reply README.md]: Oops done

Comment (imphil) [reply README.md]: Oh no worries, you couldn't have known. Thanks for the update!

**Review by imphil — APPROVED** [review]
Looks good, thanks for your contribution!

**Comment by cmarqu** [comment]
Do we maybe want to deprecate the old names first, using code like https://github.com/cocotb/cocotb/pull/4189/files#diff-4c6ad64db275846a60679531019bc2ff636c9a98c68a4878efde1d0271a7ad65R25-R30 ?

**Comment by ktbarrett** [comment]
@cmarqu This has already been done.
